chain-signatures/node/src/indexer_eth.rs

```diff
@@ -477,12 +492,24 @@ pub async fn run(
     app_data_storage: AppDataStorage,
     node_near_account_id: AccountId,
     backlog: Backlog,
+    mut contract_watcher: ContractStateWatcher,
+    mesh_state: watch::Receiver<MeshState>,
+    node_client: NodeClient,
 ) {
     let Some(eth) = eth else {
         tracing::warn!("ethereum indexer is disabled");
         return;
     };
 
+    // Wait for threshold to be available
+    let threshold = contract_watcher.wait_threshold().await;
+    if threshold > 0 {
+        let mesh_state = mesh_state.borrow().clone();
+        backlog
+            .recover(&mesh_state, &node_client, threshold, &[Chain::Ethereum])
+            .await;
+    }
+
```

Comment:
Not necessarily for this PR, but I wonder, does it make sense to unconditionally try and recover from remote-fetched checkpoint?

What if we do a quick reboot, wouldn't we have local data that's more recent? Maybe we can check how far we are behind the latest block height, to see if we even need to catch up with remote checkpoints?